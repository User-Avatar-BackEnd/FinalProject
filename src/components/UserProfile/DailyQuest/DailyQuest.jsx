import React from 'react';
import classNames from 'classnames';

import styles from './DailyQuest.module.scss';

const DailyQuest = ({ title, isCompleted }) => {

  return (
    <div className={styles.DailyQuest}>
      <div className={styles.header}>
        <h3>DAILY QUEST</h3>
      </div>
      <div className={classNames(styles.quest, isCompleted ? styles.completed : '')}>
        <div>
          <h4>{title}</h4>
          <span>x2 points!</span>
        </div>
        {isCompleted &&
          <p className={styles.completedMessage}>Completed</p>
        }
      </div>
    </div>
  );

}

export default DailyQuest;
