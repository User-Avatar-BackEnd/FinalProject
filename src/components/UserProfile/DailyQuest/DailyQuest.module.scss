@font-face {
  font-family: 'Joystix';
  src : url('../../../fonts/joystix monospace.ttf');
}

.DailyQuest {
  text-align: center;
  font-family: 'Joystix', sans-serif ;
  margin-top: 50px;

  & .header {

    & h3 {
      font-size: 36px;
      margin: 5px;
    }
  }

  & .quest {
    --border-width: 3px;

    position: relative;
    width: 350px;
    min-height: 200px;
    margin: 0 auto;
    padding: 10px;
    font-size: 20px;
    background: #FDFDFF;
    box-sizing: border-box;
    cursor: pointer;
    border-radius: var(--border-width);

    &:not(.completed)::after {
      position: absolute;
      content: "";
      top: calc(-1 * var(--border-width));
      left: calc(-1 * var(--border-width));
      z-index: -1;
      width: calc(100% + var(--border-width) * 2);
      height: calc(100% + var(--border-width) * 2);
      background: linear-gradient(
                      60deg,
                      hsl(224, 85%, 66%),
                      hsl(269, 85%, 66%),
                      hsl(314, 85%, 66%),
                      hsl(359, 85%, 66%),
                      hsl(44, 85%, 66%),
                      hsl(89, 85%, 66%),
                      hsl(134, 85%, 66%),
                      hsl(179, 85%, 66%)
      );
      background-size: 300% 300%;
      background-position: 0 50%;
      border-radius: calc(2 * var(--border-width));
      animation: moveGradient 4s alternate infinite;
    }

    @keyframes moveGradient {
      50% {
        background-position: 100% 50%;
      }
    }

    &.completed {
      border: 3px solid #bbbbbb;
      border-radius: 10px;

      & div {
        text-decoration: line-through;
        color: #bbbbbb;
      }
    }

    & .completedMessage {
      color: darkgreen;
      font-size: 28px;
    }
  }
}