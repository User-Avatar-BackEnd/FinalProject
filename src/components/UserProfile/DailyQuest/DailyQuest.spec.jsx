import React from 'react';
import { shallow } from 'enzyme';

import DailyQuest from './DailyQuest';

describe('DailyQuest component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DailyQuest />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});