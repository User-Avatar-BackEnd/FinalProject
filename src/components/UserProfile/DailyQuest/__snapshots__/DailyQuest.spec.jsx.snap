// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyQuest component should match snapshot 1`] = `
<div
  className="DailyQuest"
>
  <div
    className="header"
  >
    <h3>
      DAILY QUEST
    </h3>
  </div>
  <div
    className="quest"
  >
    <div>
      <h4 />
      <span>
        x2 points!
      </span>
    </div>
  </div>
</div>
`;
